master-server/src/main/java/com/replicated_log/master_server/service/impl/BrokerServiceImpl.java

```diff
@@ -0,0 +1,132 @@
+package com.replicated_log.master_server.service.impl;
+
+import com.replicated_log.master_server.model.Address;
+import com.replicated_log.master_server.model.HealthStatus;
+import com.replicated_log.master_server.model.Item;
+import com.replicated_log.master_server.repository.BrokerRepository;
+import com.replicated_log.master_server.service.AddressService;
+import com.replicated_log.master_server.service.BrokerService;
+import com.replicated_log.master_server.service.ItemService;
+import com.replicated_log.master_server.service.MasterClient;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import java.util.Map;
+import java.util.Set;
+
+@Service
+public class BrokerServiceImpl implements BrokerService {
+
+    private final Logger LOG = LogManager.getLogger(BrokerServiceImpl.class);
+
+    @Autowired
+    private BrokerRepository brokerRepository;
+
+    @Autowired
+    private ItemService itemService;
```

Comment:
try to avoid service in service using, sooner or later it will lead to circular dependency

Reply:
tryed to avoid service in service using, this was used only for complex logic

Reply:
try to avoid service in service using, sooner or later it will lead to circular dependency
 interesting note - disputable  😃